src/App.js

```diff
@@ -2,24 +2,99 @@ import React from 'react';
 
 import './App.scss';
 
-const App = () => {
-  setInterval(() => {
-    const date = new Date();
+class App extends React.Component {
+  timerID = null;
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      date: new Date(),
+      isClockVisible: true,
+      clockName: 'clock',
+    };
+  }
+
+  componentDidMount() {
+    this.startClock();
+  }
+
+  componentWillUnmount() {
+    this.stopClock();
+  }
+
+  startClock = () => {
+    if (this.timerID === null) {
+      this.timerID = setInterval(() => {
+        this.tick();
+
+        // eslint-disable-next-line
+        console.log(this.state.date.toLocaleTimeString());
+      }, 1000);
+    }
+  }
+
+  stopClock = () => {
+    clearInterval(this.timerID);
+    this.timerID = null;
+  }
+
+  tick = () => {
+    this.setState({
+      date: new Date(),
+    });
+  }
+
+  showClock = () => {
+    this.startClock();
+    this.setState({
+      isClockVisible: true,
+    });
+  }
+
+  hideClock = () => {
+    this.stopClock();
```

Comment:
```suggestion
```
redundant. If `isClockVisible` is false, then `componentWillUnmount` will work (which will mean that your interval will be cleared)